ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -86,13 +88,20 @@ private[kvutils] class ProcessTransactionSubmission(
   ): Commit[Unit] = {
     val dedupKey = commandDedupKey(transactionEntry.submitterInfo)
     get(dedupKey).flatMap { dedupEntry =>
-      if (dedupEntry.isEmpty) {
+      def isAfterDeduplicationTime(stateValue: DamlStateValue): Boolean = {
+        lazy val cmdDedup = stateValue.getCommandDedup
+        lazy val dedupTime = parseTimestamp(cmdDedup.getDeduplicationTime).toInstant
+        !stateValue.hasCommandDedup || !cmdDedup.hasDeduplicationTime || dedupTime.isBefore(
+          Instant.now()) // should this use record time instead?
```

Comment:
Yes, we must always use record time, local clock should not be trusted and this would be non-deterministic with distributed validators.

Reply:
Thought so, thanks.